airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/NormalizationActivityImpl.java

```diff
@@ -156,28 +169,48 @@ public NormalizationSummary normalize(final JobRunConfig jobRunConfig,
         () -> context);
   }
 
+  @VisibleForTesting
+  static void activateStrictNormalizationComparisonIfPossible(final IntegrationLauncherConfig destinationLauncherConfig) {
+    // Strict comparison was branched from normalization 0.2.25, so we shouldn't apply it if we're
+    // trying to use a newer version of normalization
+    if (NON_STRICT_COMPARISON_IMAGE_TAG.equals(getNormalizationImageTag(destinationLauncherConfig))) {
+      replaceNormalizationImageTag(destinationLauncherConfig, STRICT_COMPARISON_IMAGE_TAG);
+    }
+  }
+
   @Trace(operationName = ACTIVITY_TRACE_OPERATION_NAME)
   @Override
   public NormalizationInput generateNormalizationInput(final StandardSyncInput syncInput, final StandardSyncOutput syncOutput) {
     return new NormalizationInput()
         .withDestinationConfiguration(syncInput.getDestinationConfiguration())
         .withCatalog(syncOutput.getOutputCatalog())
-        .withResourceRequirements(normalizationResourceRequirements);
+        .withResourceRequirements(normalizationResourceRequirements)
+        .withWorkspaceId(syncInput.getWorkspaceId());
   }
 
   @VisibleForTesting
   static boolean normalizationSupportsV1DataTypes(final IntegrationLauncherConfig destinationLauncherConfig) {
```

Comment:
Can you add a test for the version of the protocol we would pick with the tags related to this change? Let's make sure we are covered here and don't run into conflict with the recent changes until it gets cleaned up.

Reply:
done